app/views/directives/create-secret.html

```diff
@@ -0,0 +1,300 @@
+<ng-form name="secretForm">
+  <div for="secretType" ng-if="!type" class="form-group">
+    <label>Secret Type</label>
+    <ui-select required ng-model="newSecret.type" search-enabled="false" ng-change="newSecret.authType = secretAuthTypeMap[newSecret.type].authTypes[0].id">
+      <ui-select-match>{{$select.selected | upperFirst}} Secret</ui-select-match>
+      <ui-select-choices repeat="type in secretTypes">
+        {{type | upperFirst}} Secret
+      </ui-select-choices>
+    </ui-select>
+  </div>
+
+  <div class="form-group">
+    <label for="secretName" class="required">Secret Name</label>
+      <span ng-class="{'has-error': nameTaken}">
+        <input class="form-control"
+          id="secretName"
+          name="secretName"
+          ng-model="newSecret.data.secretName"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
```

Comment:
See

https://github.com/spadgett/origin-web-console/blob/e8e3ecdc1dbe23297674577da7180d48b0c6a3c2/app/views/create-project.html#L27-L29

for an example, but need to check the values against valid secret names, which might have different rules.